common/conf/conf.cpp

```diff
@@ -155,29 +158,7 @@ expected::ExpectedSize MenderConfig::ProcessCmdlineArgs(
 	bool help_arg = false;
 
 	CmdlineOptionsIterator opts_iter(
-		start,
-		end,
-		{
-			"--config",
-			"-c",
-			"--fallback-config",
-			"-b",
-			"--data",
-			"-d",
-			"--log-file",
-			"-L",
-			"--log-level",
-			"-l",
-			"--trusted-certs",
-			"-E",
-		},
-		{
-			"--skipverify",
-			"--version",
-			"-v",
-			"--help",
-			"-h",
-		});
+		start, end, GlobalOptsSetWithValue(), GlobalOptsSetWithoutValue());
```

Comment:
Good!

But can we also do the same with the options in `ParseAuthArguments` and `ParseUpdateArguments`? I know, I'm being annoying! 😄

While you're at it, could you also add an `else` block to the end of each flag matcher in those two functions, with an `assert(false)`, just to make sure we catch the case where a flag is not handled.

Reply:
Good suggestions, both of them.

Reply:
No, I think it's good! 👍 Now there is no duplication.